src/system-register.ts

```diff
@@ -0,0 +1,18 @@
+import {ModuleCache} from './module-cache';
+
+export class SystemRegister {
+  constructor(public cache: ModuleCache) {}
+
+  register(name: string, deps: string[], program: Function): void {
+    var module: any = {}; //blank object for module
+
+    var exports: Function = (name: string, member: any) => module[name] = member;
+    var obj: any = program(exports);
+
+    for(var i in deps) {
```

Comment:
Wrap this inside of a Promise chain. `this.cache.get(deps).then( (modules: any[]) => { ... } )`. The for loop will go inside of the braces and will iterate over modules.

Also, it will need to be a for loop, not a foreach loop. So `for (var i = 0; i < modules.length; i++) { ... }`. We need to know the position so that we can call the appropriate setter.